icu4c/source/test/intltest/locnmtst.cpp

```diff
@@ -422,4 +423,59 @@ void LocaleDisplayNamesTest::TestRootEtc() {
   delete ldn;
 }
 
+void LocaleDisplayNamesTest::VerifySubstitute(LocaleDisplayNames* ldn) {
+  UnicodeString temp;
+  // Ensure the default is UDISPCTX_SUBSTITUTE
+  UDisplayContext context = ldn->getContext(UDISPCTX_TYPE_SUBSTITUTE_HANDLING);
+  test_assert(UDISPCTX_SUBSTITUTE == context);
+  ldn->regionDisplayName("xy", temp);
+  test_assert_equal("xy", temp);
+  ldn->languageDisplayName("xy", temp);
+  test_assert_equal("xy", temp);
+  ldn->scriptDisplayName("wxyz", temp);
+  test_assert_equal("wxyz", temp);
+  ldn->variantDisplayName("abc", temp);
+  test_assert_equal("abc", temp);
+  ldn->keyDisplayName("efg", temp);
+  test_assert_equal("efg", temp);
+  ldn->keyValueDisplayName("ca", "ijk", temp);
+  test_assert_equal("ijk", temp);
+}
+
+void LocaleDisplayNamesTest::VerifyNoSubstitute(LocaleDisplayNames* ldn) {
```

Comment:
Please add a test for some key that is present in root but not in the language bundle.

Reply:
I believe no such thing exist
see https://github.com/unicode-org/icu/blob/master/icu4c/source/data/lang/root.txt
and
https://github.com/unicode-org/icu/blob/master/icu4c/source/data/region/root.txt

Reply:
OK.